.ci/docker/build.sh

```diff
@@ -41,6 +41,8 @@ BUILD_DOCS=1
 cp ../../requirements-lintrunner.txt ./
 
 # Copy arm setup script from root to here
+# TODO(huydhn): Figure out a way to rebuild the Docker image automatically
+# with a new image hash when the content here is updated
 cp -r ../../examples/arm/ ./arm
```

Comment:
@huydhn IIUC, it will re-run this step (copy new setup.sh and its deps) when the docker image rebuild is triggered, in this case when Arm bumps up the reference model version in setup.sh. So if we are tracking the changes in setup.sh, we should get the automatic docker image rebuild as needed, right? Just curious if you indicate something different

Reply:
Yeah, the rebuild is done alright, but there is a complication where the Docker image hash tag is computed from the commit hash of the `.ci/docker` directory, the standard location of everything docker-related.  So, what we have here is that the Docker image is rebuilt, but its hash tag remains the same because `examples/arm/` is not under `.ci/docker`, so the job doesn't reupload the image thinking that nothing has changed.  That's the reason why @Jerry-Ge tried to update the branch, but didn't see the new change took effect.

I could apply the same workaround that PyTorch is using here by removing `cp` and creating a softlink.

Reply:
Oh I see, the hash here is for the docker image not the hash of the `reference_model`. @huydhn Maybe you could elaborate more about when the docker image hash tag is checked and compared? I understand that once those files are copied, the hash of the docker image won't change, but shouldn't the docker image build be triggered based on .github/workflows/docker-builds.yml where we set to monitor the `examples/arm/` dir. In that case, why is it still needed to check and compare the docker image hash?

Reply:
The logic behind that is in this GitHub action https://github.com/pytorch/test-infra/blob/main/.github/actions/calculate-docker-image/action.yml#L161-L163, you can see that the Docker image is not pushed again if the same hash tag already exists.  And the hash tag comes from the content of `.ci/docker` folder itself https://github.com/pytorch/test-infra/blob/main/.github/actions/calculate-docker-image/action.yml#L70

Reply:
I think this option should help https://github.com/pytorch/test-infra/pull/4866, this force pushes the rebuilt Docker image even if the hash tag already exists.